hibernate-core/src/main/java/org/hibernate/event/internal/DefaultMergeEventListener.java

```diff
@@ -318,12 +318,26 @@ protected void entityIsDetached(MergeEvent event, MergeContext copyCache) {
 		final Object result = source.getLoadQueryInfluencers().fromInternalFetchProfile(
 				CascadingFetchProfile.MERGE,
 				() -> source.get( entityName, clonedIdentifier )
-
 		);
+
 		if ( result == null ) {
-			//TODO: we should throw an exception if we really *know* for sure
-			//      that this is a detached instance, rather than just assuming
-			//throw new StaleObjectStateException(entityName, id);
+			// a timestamped version would not enter here in a new creation
+			if ( persister.isVersioned() ) {
+				Object versionValue = persister.getValue( entity, persister.getVersionProperty() );
+				if ( versionValue != null ) {
+					boolean isPrimitiveVersion = persister.getVersionMapping()
+							.getVersionAttribute()
+							.getPropertyAccess()
+							.getGetter()
+							.getReturnTypeClass()
+							.isPrimitive();
+					// IF primitive AND defaultvalue == versionvalue 0 => DON'T throw
+					// TODO evaluate if the primitive version types can be seeded to 1 (see also StaleObjectMergeTest.testStalePrimitiveAndWrapperVersionEntityMerged()
```

Comment:
Please remove the todo, as it is not actionable here. We need to discuss changing the seed value separately.